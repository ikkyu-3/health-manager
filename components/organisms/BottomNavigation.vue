<template>
  <v-bottom-navigation v-model="bottomNav" fixed grow>
    <navigation-button
      v-for="(button, index) in buttons"
      :key="index"
      :to="button.to"
      :value="button.value"
      :text="button.text"
      :icon="button.icon"
    />
  </v-bottom-navigation>
</template>

<script lang="ts">
import { createComponent } from '@vue/composition-api'
import NavigationButton from '@/components/molecules/button/NavigationButton.vue'

export default createComponent({
  components: { NavigationButton },
  setup() {
    const buttons = [
      {
        to: 'workouts',
        value: 'workouts',
        text: 'Workouts',
        icon: 'fa-dumbbell'
      },
      {
        to: 'schedule',
        value: 'schedule',
        text: 'Schedule',
        icon: 'fa-calendar-alt'
      },
      {
        to: 'workout-data',
        value: 'workout-data',
        text: 'Workout Data',
        icon: 'fa-list-alt'
      },
      {
        to: 'health-planet',
        value: 'health-planet',
        text: 'Health Planet',
        icon: 'fa-globe'
      }
    ]

    const bottomNav: string = buttons[0].value

    return {
      buttons,
      bottomNav
    }
  }
})
</script>

<style></style>
