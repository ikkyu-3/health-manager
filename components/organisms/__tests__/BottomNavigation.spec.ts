import Vuetify from 'vuetify'
import { mount, createLocalVue } from '@vue/test-utils'
import BottomNavigation from '@/components/organisms/BottomNavigation.vue'

const localVue = createLocalVue()

describe('organisms/BottomNavigation.vue', () => {
  let vuetify: typeof Vuetify

  it('snapshot', () => {
    const wrapper = mount(BottomNavigation, { localVue, vuetify })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
