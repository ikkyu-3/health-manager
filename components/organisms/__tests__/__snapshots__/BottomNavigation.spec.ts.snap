// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisms/BottomNavigation.vue snapshot 1`] = `"<div class=\\"v-bottom-navigation v-item-group theme--light v-bottom-navigation--grow v-bottom-navigation--fixed v-bottom-navigation--shift\\" style=\\"height: 56px; transform: none; background-color: rgb(25, 118, 210); border-color: #1976d2; color: rgb(250, 250, 250); caret-color: #fafafa;\\"><a value=\\"workouts\\" class=\\"v-btn v-btn--contained v-btn--router theme--light v-btn--active v-size--default\\"><span class=\\"v-btn__content\\"><span class=\\"button-text\\">Workouts</span> <i aria-hidden=\\"true\\" class=\\"v-icon notranslate fa fa-dumbbell theme--light\\"></i></span></a><a value=\\"schedule\\" class=\\"v-btn v-btn--contained v-btn--router theme--light v-size--default\\"><span class=\\"v-btn__content\\"><span class=\\"button-text\\">Schedule</span> <i aria-hidden=\\"true\\" class=\\"v-icon notranslate fa fa-calendar-alt theme--light\\"></i></span></a><a value=\\"workouts-data\\" class=\\"v-btn v-btn--contained v-btn--router theme--light v-size--default\\"><span class=\\"v-btn__content\\"><span class=\\"button-text\\">Workouts Data</span> <i aria-hidden=\\"true\\" class=\\"v-icon notranslate fa fa-list-alt theme--light\\"></i></span></a><a value=\\"health-planet\\" class=\\"v-btn v-btn--contained v-btn--router theme--light v-size--default\\"><span class=\\"v-btn__content\\"><span class=\\"button-text\\">Health Planet</span> <i aria-hidden=\\"true\\" class=\\"v-icon notranslate fa fa-globe theme--light\\"></i></span></a></div>"`;
